src/Renci.SshNet/ShellStream.cs

```diff
@@ -550,6 +575,9 @@ public string Expect(Regex regex, TimeSpan timeout)
         {
             var text = string.Empty;
 
+            //  Flush the write buffer in case expecting output from previous write.
+            if (!_isDisposed) Flush();
```

Comment:
This goes back to the "half-used"/"half-Disposed" issue when an abrupt channel closure causes the ShellStream to abruptly be in a Disposed state without explicit user interaction. Under normal API usage, I agree 100% - Disposed state should be private, calling methods on a Disposed object should throw an *ObjectDisposedException*, and all of this behavior is appropriate because it is up to the user to set the 'disposed' state alone. Essentially, the user caused the Disposal, so therefore should know better than to use an object he/she disposed.

In this object, that behavior is not present. Maybe I should make a new private variable that specifies if the ShellStream is in a 'half-open' state? If so, how should the *Write* methods behave under this state? I'm genuinely curious about your thoughts. Thanks!